### PR TITLE
planner: add warning if fallback to static plan from dynamic

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -4408,6 +4408,10 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		if usePartitionProcessor {
 			b.optFlag = b.optFlag | flagPartitionProcessor
 			b.ctx.GetSessionVars().StmtCtx.UseDynamicPruneMode = false
+			if isDynamicEnabled {
+				b.ctx.GetSessionVars().StmtCtx.AppendWarning(
+					fmt.Errorf("disable dynamic pruning due to %s has no global stats", tableInfo.Name.String()))
+			}
```
Comment (xuyifangreeneyes) [inline planner/core/logical_plan_builder.go:4414]: Do we need a test case to test the warning?

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

**Review by hawkingrei — APPROVED** [review]

File: planner/core/logical_plan_builder.go (hunk 1 above)
Comment (Yisaer) [reply planner/core/logical_plan_builder.go:4414]: I think not

**Pull request merged**